Gems/AtomLyIntegration/CommonFeatures/Code/Source/PostProcess/EditorModeFeedback/EditorEditorModeFeedbackSystemComponent.cpp

```diff
@@ -0,0 +1,268 @@
+                    }
+                }
+
+                m_enabled = true;
+            }
+        }
+
+        void EditorEditorModeFeedbackSystemComponent::OnEditorModeDeactivated(
+            [[maybe_unused]] const AzToolsFramework::ViewportEditorModesInterface& editorModeState,
+            AzToolsFramework::ViewportEditorMode mode)
+        {
+            if (mode == AzToolsFramework::ViewportEditorMode::Focus)
+            {
+                m_enabled = false;
+            }
+        }
+
+        void EditorEditorModeFeedbackSystemComponent::OnTick([[maybe_unused]] float deltaTime, [[maybe_unused]] AZ::ScriptTimePoint time)
+        {
+            if (!m_enabled)
+            {
+                return;
+            }
+
+            const auto focusModeInterface = AZ::Interface<AzToolsFramework::FocusModeInterface>::Get();
+            if (!focusModeInterface)
+            {
+                return;
+            }
+
+            // TODO: see if there is a more reliable method of creating the required resources once the depender systems are initialized
+            if (!m_maskMaterial)
+            {
+                m_maskMaterial = CreateMaskMaterial();
+            }
+
+            // Build the draw packets (where required) for each registered component and add them to the draw list
+            const auto focusedEntityIds = focusModeInterface->GetFocusedEntities(AzToolsFramework::GetEntityContextId());
+            for (const auto& focusedEntityId : focusedEntityIds)
+            {
+                const auto entityComponentMeshHandleDrawPacketsIterator = m_entityComponentMeshHandleDrawPackets.find(focusedEntityId);
+                if (entityComponentMeshHandleDrawPacketsIterator == m_entityComponentMeshHandleDrawPackets.end())
+                {
+                    // No drawable data for this entity
+                    continue;
+                }
+
+                auto& [entityId, componentMeshHandleDrawPackets] = *entityComponentMeshHandleDrawPacketsIterator;
+
+                for (auto& [componentId, meshHandleDrawPackets] : componentMeshHandleDrawPackets)
+                {
+                    const auto scene = RPI::Scene::GetSceneForEntityId(entityId);
+                    if (meshHandleDrawPackets.m_meshDrawPackets.empty())
+                    {
+                        if (const auto featureProcessor =
+                                RPI::Scene::GetFeatureProcessorForEntity<MeshFeatureProcessorInterface>(focusedEntityId))
+                        {
+                            const auto maskMeshObjectSrg =
+                                CreateMaskShaderResourceGroup(featureProcessor, *meshHandleDrawPackets.m_meshHandle, m_maskMaterial);
+                            const auto view = GetViewFromScene(scene);
+                            const auto worldTM = GetWorldTransformForEntity(focusedEntityId);
+                            meshHandleDrawPackets.m_meshDrawPackets = BuildMeshDrawPackets(
```

Comment:
While this is not extremely expensive, it can be avoided by holding a map of draw packets per entity together with the lod index, and only if the lod index changes, throw away the existing and create a new packet with the current lod mesh.
Since this is editor mode I'm less concerned but definitely a point to improve instead of doing it every frame for all selected models

Reply:
Yeah, that's got to be the way forwards, i think. I will hold back on this specific change and consider it a limitation of the current implementation because I think this sort of detail work to really lock in full support for the mesh component would be better served as part of the wider architectural shift when we move the rendering logic over to a feature processor 👍